coreos/tectonic-installer
# installer/tests/sanity: add a test that ensure all manifests are created

This PR makes the sanity tests verify that all the resources defined by the generated manifests are created and present on the provisioned clusters. This aims at avoiding situations where we thought we were deploying Prometheus, while we were not due to a bug.

The PR is huge because we *had* to update the vendors. They were from mid-2016 and because we are now using a `kubectl` command, conflicts arise as the different packages in Kubernetes were not using the same client. Now, everything uses `client-go`. **You want to look at the third commit.** If you prefer, I can make a PR that is only doing the vendor bump, wait for it to be merged, and rebase.

Fixes https://github.com/coreos/tectonic-installer/issues/1020
Needs to be updated after https://github.com/coreos/tectonic-installer/issues/1037 (set of experimental manifests)

**How to test?**
```
export MANIFEST_EXPERIMENTAL=false
export MANIFEST_PATHS=build/<CLUSTER>/generated/
export TEST_KUBECONFIG=build/<CLUSTER>/generated/auth/kubeconfig

go test -v github.com/coreos/tectonic-installer/tests/smoke -cluster
```

You can toggle the `MANIFEST_EXPERIMENTAL` flag to generate a failure depending on whether you deployed a clluster with experimental features or not. You may also create extra or wrong manifests in the generated manifest folder and see the test complaining because the manifest is not there, or cannot be parsed, respectively.
